src/Material.h

```diff
@@ -43,6 +45,13 @@
 int material_id();
 void reset_material_id();
 
+enum materialType {
+  GENERIC_MATERIAL,
```

Comment:
Perhaps simplify these as ``GENERIC``, ``MACRO``, ``ISO`` (or ``MICRO``?) and ``SUBGROUP``?